README.md

```diff
@@ -0,0 +1,180 @@
+
+```
+import {GraphQLInt} from 'graphql';
+
+// Foo.ts
+export default new NodeType({
+    // this determines the GraphQL type name
+    name: 'Foo',
+
+    // this provides graphql documentation on this type
+    description: 'Foo is the catch all example for all things programming',
+
+    // returns an object containing all of the fields for this graphql type
+    fields() {
+        return {
+            bar: {
+                type: GraphQLInt,
+                description: 'A unit of Foo'
+            }
+        }
+    },
+
+    createResource(ctx) {
+        // as long as the `/foo` endpoint follows standard CRUD, then HttpResource should be able
+        // to handle fetching it correctly
+        return new HttpResource({
+            endpoint: "/foo"
+        })
+    }
+});
+
+
+```
+
+
+
+### Adding the Node Field to Your Query Object
+
+In order to fulfill the contract required by Relay, you need to expose a field `node` on your `Query` type.
+
+```
+// query.ts
+import { getConfig } from '@4c/graphql-node-resource/config';
+
+const config = getConfig();
+
+export default new GraphQLObjectType({
+  name: 'Query',
+
+  fields: {
+    node: config.nodeField,
+    nodes: config.nodesField,
+  },
+})
+```
+
+This will expose `node` and `nodes` on your Query graphql type.
+
+### Resolving a Node Type
+
+There are two ways to resolve a Node type from a parent type:
+1. use the `NodeType.getResource(context)` function to get the HTTPResource object and use that to get the node
+2. resolve the field to an object containing an id such as `{id: 'foo123'}
+
+### Using Get Resource
+
+You can resolve the resource directly by getting the HTTPResource object from the NodeType instance:
+```
+// Baz.ts
+export default new GraphQLObjectType({
+    name: 'Baz',
+    fields: {
+        foo: {
+            type: new GraphQLList(Foo),
+            resolve(context) {
+                const resource = Foo.getResource(context) as HTTPResource;
+                return resource.get(); // get a list of Foos
+            }
+        }
+    }
+})
+```
+
+This is useful when you don't have an ID and need to resolve a list of resources or
+fetch a list of resources and then find the object you are looking for.
+
+### Passing an object with an id
+
+The `NodeType` class provides default resolvers for all defined fields. These resolvers will try to fetch the Node from the provided resource and then resolve to the matching field in the response. In order to do this, the `id` field must be present in the object that is being resolved.
+
+In the example below, the resolver for the `foo` field returns an object with an `id` property. The default resolvers provided by the NodeType will use the `id` property to try and fetch the `Foo` resource.
+
+```
+// Baz.ts
+export default new GraphQLObjectType({
+    name: 'Baz',
+    fields: {
+        foo: {
+            type: Foo,
+            resolve() {
```

Comment:
Might as well add the standard args to the resolve function here, since they're always the same: `(parent, args, context, info)`

Reply:
```suggestion
            resolve: (parent, args, context, info) {
```